### ISSUE TITLE
Accessibility Issues with Canvas: Report Attached

### ISSUE THREAD

**Issue by amcallisterocad** [instructure/canvas-lms#201]
Please review this comprehensive accessibility report by Greg Gay from the IDRC http://idrc.ocad.ca

http://webspace.ocad.ca/~amcallister/idrc/WCAG2_review_Canvas_lms.pdf

The General Overview of his report follows:

> General Comments
> This evaluation looks at the accessibility and usability of the Student facing components of the Canvas LMS. It is not an exhaustive review of all features, but rather more general coverage of the accessibility and usability of the student features as a whole.
> 
> A number of significant barriers were identified that would effectively make parts of Canvas unusable by some Assistive Technology (AT) users. 
> 
> Many features were identified that would not operate by keyboard. The Inbox for example, keyboard users could not select a user to send to, could not archive or delete messages, and could not display a thread of messages, all of which require a mouse click to operate. These problems can often be resolved by ensuring event handlers are device independent, rather than using those that rely on mouse input.
> 
> In other cases content could not be read by AT. The context sensitive help that opens in a modal dialog box is unreadable by AT. These problems can often be resolved by sending focus to the beginning of a dialog box when it opens, and when closed redirecting focus back to the position where the dialog was opened.
> 
> The dynamically presented feedback messages, either next to a form input incorrectly entered, or dropped down from the top of the screen, are not readable by AT. Using WAI-ARIA Live Regions may be a way to maintain dynamic messaging, having messages announce when they are injected into a page.
> 
> In places a control is used to open a feature in another area of the screen without notifying the user. For example, when a person clicks the “How do I…” link in the right column of the Calendar, an area opens at the bottom of the screen without warning. When clicking “write a reply…” in a discussion post, an “Insert Content…” area opens off to the right without warning.  These features are likely to go unnoticed by many AT users. Ideally controls open features immediately following the control, though strategically sending focus to a remotely opened feature may also help make such features more accessible. In some cases, live regions could also be used to announce features injected into a remote area of a page.
> 
> Another significant usability issue is the positioning of keyboard focus when a feature is opened or closed, often sent back to the start of the page. When a Recent Activity thread is opened on the Dash Board, after an event is added to the calendar, or when opening a folder in the Files manager, among others, keyboard focus returns to the start of the page, requiring much effort to return to where one left off. These problems can often be resolved by scripting focus to redirect back to the position where an action was initiated.
> 
> In places throughout the review, “Fail?” or “Pass?” have been used where a fail or pass is questionable. “Pass?” is used in places where a single instance of a barrier has been identified, perhaps an oversight, or where it could be argued that an item might fail or pass, typically a minor issue, leaning toward a Pass. “Fail?” is used in cases where an item could be argued as a fail or pass, leaning toward a fail. In all cases, developers should consider the recommendations made to remove any potential argument.
